infra/modules/network/data/main.tf

```diff
@@ -3,30 +3,34 @@ module "interface" {
   name   = var.name
 }
 
+module "project_config" {
```

Comment:
as we discussed in our 1:1 don't introduce dependencies on config modules in reusable modules

Reply:
I can definitely make that change! I just totally totally remember the motivation there. Like, that's the conceptual difference between nesting the network model and nesting the project config module?

Reply:
infra/modules/ are reusable modules. they should have no dependencies on anything project specific.
Longer term once these modules are stable we may decided to [publish them](https://developer.hashicorp.com/terraform/registry/modules/publish) to the terraform registry, so we should have no dependencies on any local project configuration.

Reply:
Ah! So infra/modules are truly meant to be as flexible as terraform modules you would find on the registry. Gotcha. 

That changes my point of view here a bit... but I'm going to continue on with nesting the VPC module

Reply:
Yeah that's also why I was on the fence about nesting the network module, because by nesting the network module it means that database and service modules now have a dependency on the network module, meaning that in order for someone to use the database and service modules they would also need to use the network module, or at least adhere to the same interface for the network module. But after seeing how it cleans up the interface I think maybe the benefit is worth the tradeoff.